src/rgw/rgw_sal_motr.cc

```diff
@@ -1378,9 +1462,31 @@ void MotrObject::gen_rand_obj_instance_name()
 {
   enum {OBJ_INSTANCE_LEN = 32};
   char buf[OBJ_INSTANCE_LEN + 1];
-
+  std::string version_id;
   gen_rand_alphanumeric_no_underscore(store->ctx(), buf, OBJ_INSTANCE_LEN);
-  key.set_instance(buf);
+  //TODO: Handle null version object case in PutObj operation.
+  // As the version ID timestamp is encoded in Base62, the maximum value
+  // for 8-characters is 62^8 - 1. This is the maximum time interval in ms.
+  constexpr uint64_t max_ts_count = 218340105584895;
+  constexpr size_t ts_len = 8, rand_id_len = 24;
+  using UnsignedMillis = std::chrono::duration<uint64_t, std::milli>;
+  const auto ms_since_epoch = std::chrono::time_point_cast<UnsignedMillis>(
+                              std::chrono::system_clock::now()).time_since_epoch().count();
+  uint64_t cur_time = max_ts_count - ms_since_epoch;
+  auto version_ts = base62_encode(cur_time, ts_len);
+  // encode random uuid string.
+  auto uuid_str = base64_encode(reinterpret_cast<const unsigned char*>(buf), rand_id_len);
+  // find and replace all "=" with "";
+  std::string search_str = "=", replace_str = "";
+  for (size_t pos = uuid_str.find(search_str, 0);
+    pos != std::string::npos && search_str.length();
+    pos = uuid_str.find(search_str, pos + replace_str.length()))
+      uuid_str.replace(pos, search_str.length(), replace_str);
```

Comment:
What is the point of the base64 encoding? According to [this](https://github.com/Seagate/cortx-rgw/blob/cbeb3efdae46f41a98845df3f3d6367fbd97eeca/src/common/random_string.cc#L98), you are already generating a random value that is URL-safe, and is the same alphabet as your base64 encoder. You can then just get your 23-bytes of random data, combine with the 8-byte timestamp, and get your 31-byte version ID. Then you don't need to truncate it, or worry about padding characters.